scipy/optimize/_highs/meson.build

```diff
@@ -264,6 +264,7 @@ _highs_wrapper = py3.extension_module('_highs_wrapper',
     '-Wno-format-truncation',
     '-Wno-non-virtual-dtor',
     '-Wno-class-memaccess',
+    '-Wp,-w',  # __STDC_FORMAT_MACROS redefinition warning.
```

Comment:
I'm a bit worried about this one.  Any ideas why default cython compile misses this warning?  Is it a real bug?

Reply:
I may be dense, but I can't find `-Wp` at all in the GCC compile flags. Nor is this easy to search for - do you have a link?

Reply:
I also need to check what the warning actually is. If what "default cython compile" you mean distutils-based, then it wouldn't surprise me if the warning is actually there - the build log is so full of random warnings that it's easy to miss.

If it is new for Meson, then my guess is it's due to `-std=c99`.

Reply:
Yes, I meant new for Meson - just on the observation that the `--werror` build passed before I split out the pyx -> c transpile step.

Reply:
The `-Wp` flag has documentation [here](https://gcc.gnu.org/onlinedocs/gcc/Preprocessor-Options.html).  It just passes the subsequent `-w` flag through to the preprocessor.  The `-w` flag [suppresses all preprocessor warning messages](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html) ... :(